src/pages/auth/signup/index.tsx

```diff
@@ -0,0 +1,206 @@
+import { useEffect, useState } from "react";
+import Head from "next/head";
+import Link from "next/link";
+import { useRouter } from "next/router";
+
+import { z } from "zod";
+import { useForm } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { BASEURL } from "utils/base";
+
+import { AlertInput } from "components/forms/alert";
+import OAuthButton from "components/buttons/OAuthButton";
+import { EyeIcon, EyeSlashIcon } from "@heroicons/react/24/outline";
+
+/**
+ * Form Styling
+ */
+const labelStyle = "block mb-2 text-sm font-medium text-gray-900 dark:text-white";
+const fieldStyle = `
+  bg-gray-100 
+  border border-gray-300 
+  text-gray-900 sm:text-sm 
+  rounded-md
+  focus:bg-gray-50 focus: border-gray-500
+  block w-full p-2.5 
+  dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 
+  dark:focus:border-gray-500 dark:focus:bg-gray-600
+  dark:text-white 
+`;
+
+/**
+   * Form Validation
+   */
+const userSchema = z
+  .object({
+    firstName: z.string().max(36),
+    lastName: z.string().min(1, { message: "The last name is required." }).max(36),
+    email: z.string().email().min(1, {message: "Email address is required"}).max(36),
+    password: z.string().min(8, {message: "Password must be at least 8 characters in length"})
+      .max(60, {message: "Password can only be at most 60 characters in length"}),
+  })
+
+type FormData = z.infer<typeof userSchema>;
+
+/**
+ * Page Component
+ */
+export default function SignUp() {
+  //Hide Password element
+  const [showPassword, setShowPassword] = useState(false);
+
+  //React Form Hooks
+  const {
+    register, handleSubmit, formState: {errors},
+  } = useForm<FormData>({
+    resolver: zodResolver(userSchema),
+  });
+
+  const router = useRouter();
+
+  useEffect(() => {
+    router.prefetch("/message/signupSuccess");
+  }, []);
+
+  //Submission Logic
+  async function onSubmit (data: any) {
```

Comment:
When time permits we can add additional error handling/logging to onSubmit fn

Reply:
It is already there, just that hot-toasts or some notif library will replace the error message